mirgecom/euler.py

```diff
@@ -20,17 +22,238 @@
+    ]
+
+    c_int, c_ext = [
+        clmath.sqrt(gamma * lp / lrho)
+        for lp, lrho in [(pint, rho.int), (pext, rho.ext)]
+    ]
+
+    fspeed_int, fspeed_ext = [
+        (clmath.sqrt(np.dot(lv, lv)) + lc)
+        for lv, lc in [(v_int, c_int), (v_ext, c_ext)]
+    ]
+
+    return join_fields(fspeed_int, fspeed_ext)
+
+
+def _facial_flux(discr, w_tpair):
+
+    dim = discr.dim
+
+    rho = w_tpair[0]
+    rhoE = w_tpair[1]
+    rhoV = w_tpair[2:]
+
+    def scalevec(scalar, vec):
+        # workaround for object array behavior
+        return make_obj_array([ni * scalar for ni in vec])
+
+    normal = with_queue(rho.int.queue, discr.normal(w_tpair.dd))
+
+    # Get inviscid fluxes [rhoV (rhoE + p)V (rhoV.x.V + delta_ij*p) ]
+    qint = join_fields(rho.int, rhoE.int, rhoV.int)
+    qext = join_fields(rho.ext, rhoE.ext, rhoV.ext)
+    # - Figure out how to manage grudge branch dependencies
+    #    qjump = join_fields(rho.jump, rhoE.jump, rhoV.jump)
+    qjump = qext - qint
+    flux_int = _inviscid_flux(discr, qint)
+    flux_ext = _inviscid_flux(discr, qext)
+
+    # Lax/Friedrichs/Rusunov after JSH/TW Nodal DG Methods, p. 209
+    #    flux_jump = scalevec(1.0,(flux_int - flux_ext))
+    flux_jump = scalevec(0.5, (flux_int + flux_ext))
+    gamma = 1.4
+    # p(ideal single gas) = (gamma - 1)*(rhoE - .5(rhoV*V))
+    pint, pext = [
+        ((gamma - 1.0) * (rhoe - 0.5 * (np.dot(rhov, rhov) / lrho)))
+        for rhoe, rhov, lrho in [
+            (rhoE.int, rhoV.int, rho.int),
+            (rhoE.ext, rhoV.ext, rho.ext),
+        ]
+    ]
+    v_int, v_ext = [
+        scalevec(1.0 / lrho, lrhov)
+        for lrho, lrhov in [(rho.int, rhoV.int), (rho.ext, rhoV.ext)]
+    ]
+
+    c_int, c_ext = [
+        clmath.sqrt(gamma * lp / lrho)
+        for lp, lrho in [(pint, rho.int), (pext, rho.ext)]
+    ]
+
+    fspeed_int, fspeed_ext = [
+        (clmath.sqrt(np.dot(lv, lv)) + lc)
+        for lv, lc in [(v_int, c_int), (v_ext, c_ext)]
+    ]
+    # fspeed_int, fspeed_ext = _get_wavespeeds(discr, w_tpair)
+
+    # - Gak!  What's the matter with this next line?   (CL issues?)
+    #    lam = np.max(fspeed_int.get(),fspeed_ext.get())
+    lam = fspeed_int
+    #    print('lam shape = ',lam.shape)
+    #    print('qjump[0] shape = ',qjump[0].shape)
+    lfr = scalevec(0.5 * lam, qjump)
+
+    # Surface fluxes should be inviscid flux .dot. normal
+    # rhoV .dot. normal
+    # (rhoE + p)V  .dot. normal
+    # (rhoV.x.V)_1 .dot. normal
+    # (rhoV.x.V)_2 .dot. normal
+    nflux = join_fields(
+        [
+            np.dot(flux_jump[i * dim : (i + 1) * dim], normal)
+            for i in range(dim + 2)
+        ]
+    )
+
+    # add Lax/Friedrichs term
+    flux_weak = nflux + lfr
+
+    return discr.interp(w_tpair.dd, "all_faces", flux_weak)
+
+
+def inviscid_operator(discr, w, t=0.0, boundaries=BoundaryBoss()):
+    """
+    Returns the RHS of the Euler flow equations:
+    :math: \partial_t Q = - \\nabla \\cdot F
```

Comment:
Invalid escape seq. (Flake8?)

Reply:
I was trying to add math stuff to the comments. I am unsure how this works, or how to generate and/or view documentation from docstrings.   I will checkout Sphinx and see if it can help.

Reply:
Sphinx will help with this.

- [Grudge doc example](https://github.com/inducer/grudge/tree/master/doc)
- [Rendered output](https://documen.tician.de/grudge/) (not gleamingly polished... but it'll do to show how things look.)
- [Docs](https://www.sphinx-doc.org/en/master/)

Reply:
[Example of math rendering](https://documen.tician.de/grudge/symbolic.html#grudge.symbolic.primitives.forward_metric_derivative)

Reply:
Here's where I *should* have responded to doc stuff....

I'm responding here about documentation-related comments. I've added a bunch of junk in docstrings comments to "quickly" address the no documentation issue; I have no math markup yet. It would be nice to link the new domain/model document that will be created in Milestone1 to the documentation (and code) that implements each term - with eqns repeated in the docstring for the code construct.

Can we close up the other doc-related convos?